managedtenants/core/addons_loader/addon.py

```diff
@@ -93,12 +97,26 @@ def load_metadata(self, environment):
 
         self._validate_schema(metadata)
         self._validate_extra_resources(environment, metadata)
+        self._validate_subscription_config(metadata)
 
         if "extraResources" in metadata:
             self.extra_resources_loader = FileSystemLoader(str(metadata_dir))
 
         return metadata
 
+    def _validate_subscription_config(self, metadata):
+        if not metadata.get("config"):
+            return
+        configs_present = set(metadata["config"].keys())
+        # configs_present should be a subset of _PERMITTED_SUBSCRIPTION_CONFIGS
+        if not configs_present <= _PERMITTED_SUBSCRIPTION_CONFIGS:
```

Comment:
This is hard to read... I'd suggest `if configs_present not in _PERMITTED_SUBSCRIPTION_CONFIGS`

Reply:
Wouldn't the inverse condition read closer to the true intention? (And maybe s/item/config/)
```python
for config in configs_present:
    if config not in _PERMITTED_SUBSCRIPTION_CONFIGS:
        raise AddonLoadError(...)
```
Minor nit so please ignore if you prefer the existing logic.

Reply:
I tend to keep conditions un-nested, for readability. Personal choice really, I understand your reasoning and it would be fine as well :)

Reply:
I _really_ prefer ugly one-liners anyway :)
```python
if any(True for x in xs if x not in valid_xs):
```

Reply:
And I prefer writing declarative code over imperative :D

Reply:
Maybe I read PEPs too much... PEP20 says:

- Flat is better than nested.
- Readability counts.

That's where I come from :)

Reply:
Oh, and that topic is covered in the SREs style guide... https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#nested-conditions

feel free to contribute to it, it's a collection of practices we try to follow.

Reply:
Nice! I'll take a look. And yes definitely nested `if`s is a smelly smell indeed.